modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-renderer/src/main/resources/META-INF/resources/js/thunks/fieldChange.es.js

```diff
@@ -0,0 +1,143 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {debounce} from 'frontend-js-web';
+
+import {EVENT_TYPES} from '../actions/types.es';
+import {evaluate} from '../util/evaluation.es';
+import {PagesVisitor} from '../util/visitors.es';
+
+let REVALIDATE_UPDATES = [];
+
+const getEditedPages = ({editingLanguageId, name, pages, value}) => {
+	const pageVisitor = new PagesVisitor(pages);
+
+	return pageVisitor.mapFields(
+		(field) => {
+			if (field.name === name) {
+				return {
+					...field,
+					localizedValue: {
+						...field.localizedValue,
+						[editingLanguageId]: value,
+					},
+					value,
+				};
+			}
+
+			return field;
+		},
+		false,
+		true
+	);
+};
+
+const debounceFn = debounce((fn) => fn(), 50);
```

Comment:
Can we have a variable for the magic number, `50`?

```js
const UPDATE_DELAY_MS = 50;
```

Reply:
Sure, I forget to create variables for those magic numbers 🤪.